ibc-apps/ics721-nft-transfer/types/src/error.rs

```diff
@@ -5,17 +5,19 @@ use displaydoc::Display;
 use http::uri::InvalidUri;
 use ibc_core::channel::types::acknowledgement::StatusValue;
 use ibc_core::channel::types::channel::Order;
-use ibc_core::handler::types::error::ContextError;
-use ibc_core::host::types::error::{DecodingError, IdentifierError};
+use ibc_core::handler::types::error::HandlerError;
+use ibc_core::host::types::error::{DecodingError, HostError, IdentifierError};
 use ibc_core::host::types::identifiers::{ChannelId, PortId};
 use ibc_core::primitives::prelude::*;
 
 #[derive(Display, Debug, derive_more::From)]
 pub enum NftTransferError {
-    /// context error: `{0}`
-    ContextError(ContextError),
+    /// handler error: `{0}`
+    Handler(HandlerError),
```

Comment:
I believe we here can instead be having `Host(HostError)` but it requires we merge #1335 first, so that the proto `TryFrom` conversions get decouple from this `HandlerError`.

Reply:
There are a few places, such as when calls to `send_packet_validate` and `send_packet_execute` are made, when a `HandlerError` is returned. These would require a new variant in `NftTransferError` in order to capture if we move away from the `HandlerError` variant.

Reply:
Ah, good point. By the way, your comment gave me another idea on how we could better define handler errors. Let’s sync up on this next week.